### ISSUE TITLE
Add functionality to turn L-curve on and off

### ISSUE THREAD

**Issue by kdahlquist** [kdahlquist/GRNmap#168]
Once the L-curve branch got merged into beta, the program now runs the L-curve automatically.  We need to be able to turn this function "on" and "off".

**Comment by kdahlquist** [comment]
I want to update this issue with documentation as to how the L-curve is supposed to work.

From @bengfitzpatrick:
There is a hard-coded vector of alphas, running largest-first to smallest-last.  This vector is hardcoded within the GRNModel function. The number of alphas is hardcoded (and hence the number of points on the L-Curve is hardcoded).  (Note that it appears that @juansc and @trixr4kdz have made a shorter vector of alphas and commented out @bengfitzpatrick's original list to make it go faster.)

The GRNModel function successively creates new input files based on the input file specified by the user.  For each alpha there is a new input sheet.  The first alpha runs the GRNmap estimation and output. Output spreadsheets and output .mat files are written.

The original input sheet is used only to create the successive input sheets.  The first input sheet is a replicate of the original input sheet, with only the value of alpha and the name of the file changed.  

The second input sheet is created from the first, but replacing the initial guesses with the optimized values from the previous alpha run.  This bit is where the tedious programming is.

The graphs are written for each alpha, but at this point only the last alpha's graphs are saved (it overwrites the previous ones each time).  We need to fix this eventually, but this is a problem for later.

In terms of turning the L-curve functionality "on" and "off", @bengfitzpatrick suggests two separate driver codes.

What we are going to do for now is create a new line in the "optimization_parameters" worksheet called "L-curve".  It will take the value "0" if the L-curve won't be run and GRNmap will use the user-supplied alpha value.  The value "1" means that the L-curve routine will run as described above.

This has already been documented with a screenshot in issue #119.

**Comment by kdahlquist** [comment]
I ran an input workbook with a parameter "L_curve" = 0 in the optimization_parameters worksheet and the model ran only once.  Will @juansc please comment on the full status of this functionality in advance of our meeting on Wednesday.

**Comment by trixr4kdz** [comment]
Tests have been added to check if the parameter exists in the optimization_parameter worksheet and if the value is valid (i.e., if it is 1 or 0). Another test to check the alpha value of the last worksheet produced was also added and so a new sheet in "/perturbation_tests" was also created.

**Comment by kdahlquist** [comment]
One last task is to un-comment-out the full array of alphas in the GRNLCurve.m

Once this is done and verified, we can close this one.

**Comment by kdahlquist** [comment]
I downloaded the beta code (after adding @bengfitzpatrick's changes this morning) and uncommented the longer list of alphas for the L-curve and tried to run it.  I got an error that seems related to the output of the graphs after it completed the run for the 5th alpha in the list.  I'm trying to run it again with make_graphs=0 to see if it will complete and we can bypass this bug.

Will @trixr4kdz (or @juansc) run the full array of alphas on a test file to see if you can reproduce the bug?

**Comment by kdahlquist** [comment]
I committed the change so the data analysis team could do their work.  The model was still running when I  left, but it had gotten past the place where it errored out.

**Comment by kdahlquist** [comment]
As per my comment to issue #172, the L-curve completed after 13 hours when make_graphs = 0.

**Comment by juansc** [comment]
@kdahlquist How many genes did the 13 hour run is?

**Comment by kdahlquist** [comment]
The 13-hour run was for a network with 15 genes and 23 edges, estimating all parameters, w, b, and P.  I'm going to initiate runs where w and P are estimated and b is fixed and where only w is estimated.  I'll post a comment with the timing, but given the potential length of the run, if I start it now, I won't see it finish until tomorrow morning.

**Comment by juansc** [comment]
Since @kdahlquist already uncommented the longer alpha list we should review this issue and close it.

**Comment by kdahlquist** [comment]
The L-curve has completed when estimating w and P, but not b with the full array of alphas.  It errored out when only estimating w and not P or b.  I will open another issue for that and close this one because this one just refers to turning the L-curve on and off, which has been demonstrated to work.
